create_base_image/create_base_image.py

```diff
@@ -0,0 +1,182 @@
+from libcloud.compute.types import Provider
+from libcloud.compute.providers import get_driver
+import os
+import time
+import argparse
+
+# Parse flag arguments
+
+parser = argparse.ArgumentParser()
+
+def add_flag(flag_name, default):
+    parser.add_argument(f'--{flag_name}', dest=flag_name,
+                        action='store', default=default)
+
+# Base image info
+add_flag('source_image_family', 'debian-10')
+add_flag('source_image_project', 'debian-cloud')
+
+# Cuttlefish repo info
+add_flag('repository_url', 'https://github.com/google/android-cuttlefish.git')
+add_flag('repository_branch', 'main')
+
+# Build artifacts info
+add_flag('build_branch', 'aosp-master')
+add_flag('build_target' ,'aosp_cf_x86_phone-userdebug')
+add_flag('build_id', '')
+
+# Build instance info
+add_flag('build_instance', 'halyard-build')
+add_flag('build_zone', 'europe-west4-a')
+parser.add_argument('--tag', action='append', dest='tags', default=[])
+
+# New image info
+add_flag('dest_image', '')
+add_flag('dest_family', '')
+add_flag('image_disk', 'halyard-image-disk')
+parser.add_argument('--respin', action='store_true', default=False)
+
+# GCE Project info
+add_flag('datacenter', 'us-central1-b')
+add_flag('project', 'cloud-android-testing')
+
+args = parser.parse_args()
+
+
+# Get GCE Driver
+ComputeEngine = get_driver(Provider.GCE)
+driver = ComputeEngine('', '',
+                       datacenter=args.datacenter, project=args.project)
+
+
+def fatal_error(msg):
+    print(f'Error: {msg}')
+    exit()
+
+def wait_for_instance():
+    not_running = True
+    while not_running:
+        time.sleep(5)
+        # uptime returns a value other than 0 when not successful
+        not_running = os.system(f'gcloud compute ssh {args.build_instance} --zone={args.build_zone} -- uptime')
+
+def update_dest_names():
+    os.system(f'gcloud compute scp {args.build_instance}:~/image_name_values \
+    name_values --zone={args.build_zone}')
+
+    variables = {}
+
+    with open('name_values') as f:
+        for line in f:
+            name, value = line.split('=')
+            variables[name] = value.strip()
+
+    cf_version = variables['cf_version']
+    args.build_id = variables['build_id']
+
+    os.system(f'rm name_values')
```

Comment:
Should be able to replace this with python lib call

Reply:
👌